File: app/src/main/java/com/vacral/foodapplycation/HomeActivity.java
package com.vacral.foodapplycation;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.vacral.foodapplycation.databinding.ActivityHomeBinding;

import java.util.ArrayList;
import java.util.List;

public class HomeActivity extends AppCompatActivity {
    private ActivityHomeBinding binding;
    private CategoryAdapter categoryAdapter;
    private List<Category> categories;
 private List<Food> foodList;
 private  FoodAdapter foodAdapter;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityHomeBinding.inflate(getLayoutInflater());
        EdgeToEdge.enable(this);
        setContentView(binding.getRoot());
        setupRecyclerView();
       loadCategories();
       loadFood();
       setupFoodRecyclerView();


        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

    }

    private void setupRecyclerView() {
        LinearLayoutManager layoutManager = new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false);


        binding.categoriesRecyclerView.setLayoutManager(layoutManager);
    }

    private void loadCategories() {
        categories = new ArrayList<>();
        categories.add(new Category("Burger", R.drawable.ic_burger));
        categories.add(new Category("Pizza", R.drawable.pizza_2));
        categories.add(new Category("Chicken", R.drawable.chicken_svgrepo_com));
        categories.add(new Category("Sushi", R.drawable.sushi_svgrepo_com));

        categoryAdapter = new  CategoryAdapter(categories, this);
        binding.categoriesRecyclerView.setAdapter(categoryAdapter);


    }
    private void setupFoodRecyclerView(){}
    private void loadFood() {
foodList = new ArrayList<>();
foodList.add(new Food("Salad Burger",4.5f,R.drawable.salad_burger, "12$"));
foodList.add(new Food("Pizza London",3.5f,R.drawable.pizza, "15$"));
foodList.add(new Food("Sushi California",5.0f,R.drawable.sushi, "17$"));
foodList.add(new Food("KFC chiken",5.0f,R.drawable.chiken, "9$"));

foodAdapter = new FoodAdapter(foodList);
binding.foodRecyclereview.setAdapter(foodAdapter);

    }

}

File: app/src/main/java/com/vacral/foodapplycation/HomeActivity.kt
package com.vacral.foodapplycation

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.recyclerview.widget.LinearLayoutManager
import com.vacral.foodapplycation.databinding.ActivityHomeBinding

class HomeActivity : AppCompatActivity() {
    // Используем `lateinit`, чтобы не использовать nullable типы '?' и небезопасный оператор '!!'.
    // Это говорит компилятору: "Я обещаю инициализировать это позже, до первого использования".
    private lateinit var binding: ActivityHomeBinding
    private lateinit var categoryAdapter: CategoryAdapter
    private lateinit var categories: MutableList<Category>
    private lateinit var foodList: MutableList<Food>
    private lateinit var foodAdapter: FoodAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityHomeBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setupCategoryRecyclerView()
        loadCategories()

        setupFoodRecyclerView()
        loadFood()

        // Более "котлиновский" способ установки слушателя
        ViewCompat.setOnApplyWindowInsetsListener(binding.main) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }

    private fun setupCategoryRecyclerView() {
        // Используем синтаксис свойств Kotlin для установки LayoutManager
        binding.categoriesRecyclerView.layoutManager =
            LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)
    }

    private fun loadCategories() {
        // Создаем изменяемый список с помощью `mutableListOf()`
        categories = mutableListOf(
            Category("Burger", R.drawable.ic_burger),
            Category("Pizza", R.drawable.pizza_2),
            Category("Chicken", R.drawable.chicken_svgrepo_com),
            Category("Sushi", R.drawable.sushi_svgrepo_com)
        )

        categoryAdapter = CategoryAdapter(categories)
        // Используем синтаксис свойств Kotlin для установки адаптера
        binding.categoriesRecyclerView.adapter = categoryAdapter
    }

    private fun setupFoodRecyclerView() {
        binding.foodRecyclerView.layoutManager =
            LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)
    }

    private fun loadFood() {
        foodList = mutableListOf(
            Food("Salad Burger", 4.5f, R.drawable.salad_burger, "12$"),
            Food("Pizza London", 3.5f, R.drawable.pizza, "15$"),
            Food("Sushi California", 5.0f, R.drawable.sushi, "17$"),
            Food("KFC chiken", 5.0f, R.drawable.chiken, "9$")
        )

        foodAdapter = FoodAdapter(foodList)

        binding.foodRecyclerView.adapter = foodAdapter
    }
}
